Automattic/wp-calypso
# Agency Dashboard: Organize hooks into their own files

This is purely a janitorial PR, intended to make Agency Dashboard hooks easier to maintain and reason about. It splits `client/jetpack-cloud/sections/agency-dashboard/hooks.tsx` into several files, most of which remain in a `hooks` directory at the same level. Others were moved closer to the one component that uses them.

## Proposed Changes

* Move `hooks.tsx` -> `hooks/index.ts`.
* Split each hook into its own file, keeping `index.ts` as a convenient location for multiple exports.
* Move `useDashboardAddRemoveLicense` and `useShowVerifiedBadge` under the components that call them, since they're only called in one place.
* Slightly refactor `getRejectedAndFulfilledRequests` to take advantage of `Array.map`, destructuring, etc.

## Testing Instructions

* Read through all code changes. Verify that with the exception of `getRejectedAndFulfilledRequests`, all hooks and their usages remain virtually the same apart from necessary adjustments to import lists.
* From the dashboard:
  * Toggle downtime monitoring on and off for any site by clicking the Monitor toggle button for that site. Verify this behavior is identical to production.
  * Do the same, but use bulk editing for as many sites as you like via the "Edit All" button.
  * Verify Tracks events are still sent from the Agency Dashboard (e.g., when you enable or disable Monitor)
  * Verify that for email and SMS contacts that are verified, the badge indicating that verification is still appropriately shown or hidden, as in production.
  * Select a site in the table and expand it. Verify that clicking the "Add Backup to see your backup" correctly queues a Backup license to be issued for that site, as in production.
